app/uk/gov/hmrc/helptosavefrontend/controllers/EligibilityCheckController.scala

```diff
@@ -44,55 +44,96 @@ class EligibilityCheckController  @Inject()(val messagesApi: MessagesApi,
                                             jsonSchemaValidationService: JSONSchemaValidationService,
-                                            app: Application,
+                                            enrolmentService: EnrolmentService,
+                                            val app: Application,
                                             auditor: HTSAuditor)(implicit ec: ExecutionContext)
   extends HelpToSaveAuth(app) with I18nSupport with Logging {
 
+  import EligibilityCheckController.EligibilityResult
+
   def confirmDetails: Action[AnyContent] =  authorisedForHtsWithInfo  {
     implicit request ⇒
       implicit htsContext ⇒
-        val result: EitherT[Future, EligibilityCheckError, (NINO, EligibilityCheckResult)] = for {
-          nino ← EitherT.fromOption[Future](htsContext.nino, EligibilityCheckError.NoNINO)
-          userDetailsURI ← EitherT.fromOption[Future](htsContext.userDetailsURI, EligibilityCheckError.NoUserDetailsURI(nino))
-          eligible ← helpToSaveService.checkEligibility(nino, userDetailsURI)
-          nsiUserInfo = eligible.result.map(NSIUserInfo(_))
-          _ <- EitherT.fromEither[Future](validateCreateAccountJsonSchema(nsiUserInfo)).leftMap(e ⇒ JSONSchemaValidationError(e, nino))
-          _ ← writeToKeyStore(nsiUserInfo).leftMap[EligibilityCheckError](e ⇒ KeyStoreWriteError(e, nino))
-        } yield (nino, eligible)
-
-        result.fold(
-          handleEligibilityCheckError, { case (nino, eligibility) ⇒
-            eligibility.result.fold {
-              auditor.sendEvent(new EligibilityCheckEvent(nino, Some("Unknown eligibility problem")))
-              SeeOther(routes.EligibilityCheckController.notEligible().url)
-            } { info ⇒
-              auditor.sendEvent(new EligibilityCheckEvent(nino, None))
-              Ok(views.html.register.confirm_details(info))
-            }
+        val eligibilityCheck = for {
+          nino            ← EitherT.fromOption[Future](htsContext.nino, NoNINO)
+          enrolmentStatus ← enrolmentService.getUserEnrolmentStatus(nino).leftMap(e ⇒ EnrolmentCheckError(nino, e))
+          result          ← handleEnrolmentStatus(enrolmentStatus, nino, htsContext)
+        } yield result
+
+        eligibilityCheck.fold(
+          handleEligibilityCheckError, {
+            case EligibilityResult(nino, Left(EnrolmentStore.Enrolled(itmpHtSFlagSet))) ⇒
+              // if the user is enrolled but the itmp flag is not set then just
+              // start the process to set the itmp flag here without worrying about the result
+              if (!itmpHtSFlagSet){
+                enrolmentService.setITMPFlag(nino).fold(
+                  e ⇒ logger.warn(s"Could not start process to set ITMP flag for user $nino: $e"),
+                  _ ⇒ logger.info(s"Process started to set ITMP flag for user $nino")
+                )
+              }
+              Ok("You've already got an account - yay!")
```

Comment:
i understood this is temporary until NSI manage account is ready , after that we redirect user to thier page from here